walk-in-interview/src/main/java/com/google/utils/ServletUtils.java

```diff
@@ -0,0 +1,124 @@
+package com.google.utils;
+
+import com.google.gson.Gson;
+
+import javax.servlet.http.HttpServletRequest;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.Date;
+
+public final class ServletUtils {
+    private ServletUtils() {}
+
+    /**
+     * @return the value of parameter with the {@code name} in the {@code request}
+     *         or returns {@code defaultValue} if that parameter does not exist.
+     */
+    public static String getStringParameter(HttpServletRequest request, String name, String defaultValue) {
+        String value = request.getParameter(name);
+        if (value == null || value.isEmpty()) {
+            return defaultValue;
+        }
+        return value;
+    }
+
+    /**
+     * Gets the integer parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed integer or default value if exception occur
+     */
+    public static int getIntParameter(HttpServletRequest request, String parameterName, int defaultValue) {
+        String resultStr = request.getParameter(parameterName);
+
+        try {
+            int result = Integer.parseInt(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO: error handling
+
+            // TODO: add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the float parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed float or default value if exception occur
+     */
+    public static float getFloatParameter(HttpServletRequest request, String parameterName, float defaultValue) {
+        String resultStr = request.getParameter(parameterName);
+
+        try {
+            float result = Float.parseFloat(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO: error handling
+
+            // TODO: add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the LocalDate parameter from html form input.
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed date or default value if exception occur
+     */
+    public static Date getDateParameter(HttpServletRequest request,
+                                        String parameterName, Date defaultValue) {
+        String resultStr = request.getParameter(parameterName);
+
+        try {
+            Date result = new SimpleDateFormat().parse(resultStr);
```

Comment:
Pick a format that both client and server can agree with. Maybe take a look at ISO 8601 format.

Reply:
we will use "yyyy-MM-dd"